src/Interactive/Features/Interactive/Core/InteractiveHost.LazyRemoteService.cs

```diff
@@ -62,20 +62,23 @@ public void Dispose(bool joinThreads)
                     }
 
                     bool initializing = true;
-                    cancellationToken.Register(() =>
-                    {
-                        if (initializing)
+                    RemoteExecutionResult initializationResult;
+                    using (cancellationToken.Register(
+                        () =>
```

Comment:
The callback is a NOP if `initializing` is false. The using statement ensures the object is disposed at the point where it would no longer have any work to do. I'm not sure what the original intent was, since I was just preserving the original semantics. Do you have a suggested comment to add?